pkgs/by-name/li/linuxhw-edid-fetcher/package.nix

```
@@ -0,0 +1,66 @@
+{ lib
+, coreutils
+, curl
+, fetchFromGitHub
+, gawk
+, gnutar
+, stdenv
+, unixtools
+, writeShellApplication
+, nix-update-script
+, displays ? { }
+}:
+
+# Usage:
+#   let
+#     edids = (linuxhw-edid-fetcher.override {
+#       displays = {
+#         PG278Q_2014 = [ "PG278Q" "2014" ];
+#       };
+#     });
+#   in
+#     "${edids}/lib/firmware/edid/PG278Q_2014.bin";
+stdenv.mkDerivation rec {
+  pname = "linuxhw-edid-fetcher";
+  version = "unstable-2023-05-08";
+
+  src = fetchFromGitHub {
+    owner = "linuxhw";
+    repo = "EDID";
+    rev = "98bc7d6e2c0eaad61346a8bf877b562fee16efc3";
+    hash = "sha256-+Vz5GU2gGv4QlKO4A6BlKSETxE5GAcehKZL7SEbglGE=";
+  };
+
+  fetch = lib.getExe (writeShellApplication {
+    name = "linuxhw-edid-fetch";
+    runtimeInputs = [ gawk coreutils unixtools.xxd curl gnutar ];
+    text = ''
+      repo="''${repo:-"${src}"}"
+      ${builtins.readFile ./linuxhw-edid-fetch.sh}
+    '';
+  });
+
+  configurePhase = lib.pipe displays [
+    (lib.mapAttrsToList (name: patterns: ''
+      "$fetch" ${lib.escapeShellArgs patterns} > "${name}.bin"
+    ''))
+    (builtins.concatStringsSep "\n")
+  ];
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    ln -s "$fetch" "$out/bin/"
+    ${lib.optionalString (displays != { }) ''
+      install -Dm 444 *.bin -t "$out/lib/firmware/edid"
```

Most Linux CLIs (and especially GNU ones) expect the positional arguments *last.* Also, it's hard to know what the `D`, `m`, and `t` flags mean unless you're very familiar with this command, so I'd recommend using the long options (where available). Taken together:

```suggestion
      install -D --mode=444 --target-directory="$out/lib/firmware/edid" *.bin
```

(unfortunately, there's no long option for `-D`.)

---

To be honest I don't even know where the `install` command comes from, just copy pasting it from existing derivations.